app/ptxboa_functions.py

```diff
@@ -147,8 +109,10 @@ def aggregate_costs(res_details: pd.DataFrame) -> pd.DataFrame:
 
 # Settings:
 def create_sidebar(api: PtxboaAPI):
+    if "settings" not in st.session_state:
+        st.session_state["settings"] = {}
```

Comment:
Is there a specific reason you put settings in a dictionary inside session state and do not place each entry in settings in the session state as an individual key? This would make code that accesses a single setting value more concise.